packages/qwik/src/core/tests/component.spec.tsx

```diff
@@ -2365,6 +2365,18 @@ describe.each([
     expect(document.body.innerHTML).toContain('I am the innerHTML content!');
   });
 
+  it('should not throw when props are null', async () => {
+    const Child = component$((props: any) => {
+      props = null;
```

Comment:
this gets optimized into basically `<div {...null} />`. Instead do `props = globalThis.stuff ? globalThis.foo : globalThis.bar;`